pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -429,9 +430,9 @@ private static PinotResourceManagerResponse toggleSegmentsForTable(@Nonnull List
       if (state == StateType.ENABLE) {
         int instanceCount = helixResourceManager.getAllInstances().size();
         if (instanceCount != 0) {
-          timeOutInSeconds = (long) ((_offlineToOnlineTimeoutInseconds * segmentsToToggle.size()) / instanceCount);
+          timeOutInSeconds = (_offlineToOnlineTimeoutInseconds * segmentsToToggle.size()) / instanceCount;
```

Comment:
Was the previous cast to long (although incorrectly applied) to take care of overflow on multiplication?

Reply:
**_offlineToOnlineTimeoutInseconds** itself is long type